cmd/genji/insert_test.go

```diff
@@ -0,0 +1,130 @@
+		{"Non closed json stream", `{"foo":"bar"`, ``, true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			db, err := genji.Open(":memory:")
+			require.NoError(t, err)
+			defer db.Close()
+
+			err = db.Exec(`CREATE TABLE foo`)
+			require.NoError(t, err)
+			err = executeInsertCommand(db, "foo", strings.NewReader(tt.data))
+			if tt.fails {
+				require.Error(t, err)
+				return
+			}
+
+			require.NoError(t, err)
+			res, err := db.Query("SELECT * FROM foo")
+			defer res.Close()
+			require.NoError(t, err)
+
+			var buf bytes.Buffer
+			err = document.IteratorToJSON(&buf, res)
+			require.NoError(t, err)
+			require.JSONEq(t, tt.want, buf.String())
+
+		})
+	}
+
+	t.Run(`Json Array`, func(t *testing.T) {
+		const jsonArray = `
+	[
+		{"Name": "Ed", "Text": "Knock knock."},
+		{"Name": "Sam", "Text": "Who's there?"},
+		{"Name": "Ed", "Text": "Go fmt."},
+		{"Name": "Sam", "Text": "Go fmt who?"},
+		{"Name": "Ed", "Text": "Go fmt yourself!"}
+	]
+`
+		jsonStreamResult := []string{`{"Name": "Ed", "Text": "Knock knock."}`,
+			`{"Name": "Sam", "Text": "Who's there?"}`, `{"Name": "Ed", "Text": "Go fmt."}`,
+			`{"Name": "Sam", "Text": "Go fmt who?"}`,
+			`{"Name": "Ed", "Text": "Go fmt yourself!"}`}
+
+		db, err := genji.Open(":memory:")
+		require.NoError(t, err)
+		defer db.Close()
+
+		err = db.Exec(`CREATE TABLE foo`)
+		require.NoError(t, err)
+		err = executeInsertCommand(db, "foo", strings.NewReader(jsonArray))
+		require.NoError(t, err)
+		res, err := db.Query("SELECT * FROM foo")
+		defer res.Close()
+		require.NoError(t, err)
+
+		i := 0
+		_ = res.Iterate(func(d document.Document) error {
+			var buf bytes.Buffer
+			err = document.ToJSON(&buf, d)
+			require.NoError(t, err)
+			require.JSONEq(t, jsonStreamResult[i], buf.String())
+			i++
+			return nil
+		})
+	})
+
+	t.Run(`Json Stream`, func(t *testing.T) {
+		const jsonStream = `
+		{"Name": "Ed", "Text": "Knock knock."}
+		{"Name": "Sam", "Text": "Who's there?"}
+		{"Name": "Ed", "Text": "Go fmt."}
+		{"Name": "Sam", "Text": "Go fmt who?"}
+		{"Name": "Ed", "Text": "Go fmt yourself!"}
+		`
+		jsonStreamResult := []string{`{"Name": "Ed", "Text": "Knock knock."}`,
+			`{"Name": "Sam", "Text": "Who's there?"}`, `{"Name": "Ed", "Text": "Go fmt."}`,
+			`{"Name": "Sam", "Text": "Go fmt who?"}`,
+			`{"Name": "Ed", "Text": "Go fmt yourself!"}`}
+		db, err := genji.Open(":memory:")
+		require.NoError(t, err)
+		defer db.Close()
+
+		err = db.Exec(`CREATE TABLE foo`)
+		require.NoError(t, err)
+		err = executeInsertCommand(db, "foo", strings.NewReader(jsonStream))
+		require.NoError(t, err)
+		res, err := db.Query("SELECT * FROM foo")
+		defer res.Close()
+		require.NoError(t, err)
+
+		i := 0
+		_ = res.Iterate(func(d document.Document) error {
+			var buf bytes.Buffer
+			err = document.ToJSON(&buf, d)
+			require.NoError(t, err)
+			require.JSONEq(t, jsonStreamResult[i], buf.String())
+			i++
+			return nil
+		})
+
```

Comment:
This test doesn't check `i`, it is possible that the inner function is never called so `i` must be checked after the call to iterate